// Copyright (c) 2024 Advanced Micro Devices, Inc. All Rights Reserved.
syntax = "proto3";

package Schola;

import "DType.proto";

message BoxSpace {

    message BoxSpaceDimension {
        float low = 1;
        float high = 2;
    }
    repeated BoxSpaceDimension dimensions = 1;
    repeated int32 shape_dimensions = 2;
    DType dtype = 3;
}

message DiscreteSpace {
    int32 high = 1;
}

message MultiDiscreteSpace {
    repeated int32 high = 1;
}

message MultiBinarySpace {
    int32 shape = 1;
}

message Space {
    oneof space {
        BoxSpace box_space = 1;
        DiscreteSpace discrete_space = 2;
        MultiDiscreteSpace multi_discrete_space = 3;
        MultiBinarySpace multi_binary_space = 4;
        DictSpace dict_space = 5;
    }
}

message DictSpace {
    map<string,Space> spaces = 1;
}
